### PR TITLE
Re-write Beginner's Guide

### PR DESCRIPTION
**Pull request by acnagy**
See https://github.com/travis-ci/docs-travis-ci-com/issues/1145 for the discussion!

### CODE REVIEW

File: user/for-beginners.md [hunk 1]
```diff
@@ -1,57 +1,53 @@
...
-   Travis CI builds for your fork of the `travis-broken-example` repository.
+Many things can happen when Travis CI runs. First, your project is brought into into our clean environment with `git clone`, then your code is compiled. Next, any tests you have are run, and their status is checked. If tests are successful, and you have set up a deployment provider, Travis CI will deploy your application. Depending on your settings, you may be notified via email, or over other platforms such as Slack. 
 
-   > Note: You can only enable Travis CI builds for repositories you have admin access to.
+One of the advantages of CI builds, is they can execute arbitrary code if you configure the build to run a custom script. This means you can setup your own notifications, upload files output during your build (artifacts), and many other processes. Travis CI can function as a basis for general automation for your development processes. 
 
-4. Take a look at `.travis.yml`, the file which tells Travis CI what to do:
+You can also customize your build into various stages, in which later stages are dependent on the result of prior stages. See our [Build Stages](../build-stages) document for details.
```
Comment (plaindocs) [inline user/for-beginners.md]: maybe move this to job vs builds vs stages?

File: user/for-beginners.md [hunk 2]
```diff
@@ -1,57 +1,53 @@
...
-   php:
-   - 5.5
-   - 5.4
-   - hhvm
-   script: phpunit Test.php
-   ```
+### Jobs vs. builds
 
-   This file tells Travis CI that this project is written in PHP, and to test `Test.php` with phpunit against PHP versions 5.5, 5.4 and HHVM.
+The words "job" and "build" can be somewhat ambiguous in the context of CI. In the Travis CI docs and materials, they refer to the following:
```
Comment (plaindocs) [inline user/for-beginners.md]: I'd reword this to something like

In the Travis CI documentation, some common words have specific meanings: 

What do you think?

File: user/for-beginners.md [hunk 3]
```diff
@@ -1,57 +1,53 @@
...
 
-6. Fix the code by making sure that `2=1+1` in `Test.php`, commit and push to GitHub. This time, the build does not fail.
+### Infrastructure and environment notes
 
-   ```bash
-   $ git add -A
-   $ git commit -m 'Testing Travis CI: fixing the build'
-   $ git push
-   ```
+Travis CI supports a few different environment, so you have different options to use. Each job runs in exactly one of the following environments:
```
Comment (plaindocs) [inline user/for-beginners.md]: Travis CI runs your build in *one* of the following environments, each with some advantages and disadvantages:

File: user/for-beginners.md [hunk 4]
```diff
@@ -1,57 +1,53 @@
...
+title: Travis CI for Beginners
 layout: en
 permalink: /user/for-beginners/
 ---
 
-If you're not familiar with continuous integration and don't already have a repository that you want to build, this guide will show you what it is all about.
+Welcome to Travis CI! If you are new here or new to Continuous Integration (CI) in general, this doc will provide context and terminology used throughout the platform and documentation. 
 
-You need to sign up for a [GitHub account](https://github.com/) if you do not already have one.
+### What is Continuous Integration (CI)?
```
Comment (noloader) [inline user/for-beginners.md]: Recommend drop the "concept". Just use "practice":

> Continuous Integration is the practice of merging small code changes frequently and testing them quickly ...

Comment (acnagy) [inline user/for-beginners.md]: @plaindocs Right... good point. Someone around here should fix that 🙃 
@noloader Sure - makes sense. It does flow better, and it keeps the first sentence from getting too in the weeks of theory vs. implementation...

File: user/for-beginners.md [hunk 5]
```diff
@@ -1,57 +1,53 @@
...
 
-### To get started with Travis CI:
+Continuous Integration is the concept and practice of merging in small code changes frequently - rather than waiting for a large integration process at the end of a development cycle. Companies and individuals may all do this process differently, but the goal is to build healthier software by developing and testing in smaller increments. This is where Travis CI can be helpful.
 
-1. On GitHub, fork the [example PHP repository](https://github.com/plaindocs/travis-broken-example).
+Travis CI, as a CI platform, supports your development process by automatically building and testing code changes, in order to provide immediate feedback on the success of the change. Travis CI also can automate some of your development process by managing deployments and notifications.
```
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs I can't remember if we discussed this... but I think we should include reference to it. That said, I'll tighten it up a bit. Plus, the way it reads sounds like we have a separate artifacts management feature which, um, is not the case at the moment.

File: user/for-beginners.md [hunk 6]
```diff
@@ -1,57 +1,53 @@
...
+Travis CI, as a CI platform, supports your development process by automatically building and testing code changes, in order to provide immediate feedback on the success of the change. Travis CI also can automate some of your development process by managing deployments and notifications.  
 
-2. [Sign in to Travis CI](https://travis-ci.org/auth) with your GitHub account, accepting the GitHub [access permissions confirmation](/user/github-oauth-scopes).
+### CI builds and automation: building, testing, deploying
 
-3. Once you're signed in, and we've synchronized your repositories from GitHub, go to your [profile page](https://travis-ci.org/profile) and enable
-   Travis CI builds for your fork of the `travis-broken-example` repository.
+Many things can happen when Travis CI runs. First, your project is brought into into our clean environment with `git clone`, then your code is compiled. Next, any tests you have are run, and their status is checked. If tests are successful, and you have set up a deployment provider, Travis CI will deploy your application. Depending on your settings, you may be notified via email, or over other platforms such as Slack.
```
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs That's muuuch better....

File: user/for-beginners.md [hunk 7]
```diff
@@ -1,57 +1,53 @@
...
+### CI builds and automation: building, testing, deploying
 
-3. Once you're signed in, and we've synchronized your repositories from GitHub, go to your [profile page](https://travis-ci.org/profile) and enable
-   Travis CI builds for your fork of the `travis-broken-example` repository.
+Many things can happen when Travis CI runs. First, your project is brought into into our clean environment with `git clone`, then your code is compiled. Next, any tests you have are run, and their status is checked. If tests are successful, and you have set up a deployment provider, Travis CI will deploy your application. Depending on your settings, you may be notified via email, or over other platforms such as Slack. 
 
-   > Note: You can only enable Travis CI builds for repositories you have admin access to.
+One of the advantages of CI builds, is they can execute arbitrary code if you configure the build to run a custom script. This means you can setup your own notifications, upload files output during your build (artifacts), and many other processes. Travis CI can function as a basis for general automation for your development processes.
```
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs Agreed... I'm trying:
> CI builds can also automate other parts of your delivery workflow...
but not sure yet if I like it

File: user/for-beginners.md (hunk 1 above)
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs We also might have discussed this as well, but I think we should include reference to it here as part of "things you can automate", or drop it entirely? Thoughts?

File: user/for-beginners.md (hunk 2 above)
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs That's a great suggestion!

File: user/for-beginners.md (hunk 3 above)
Comment (acnagy) [inline user/for-beginners.md]: @plaindocs that's perfect!

File: user/for-beginners.md (hunk 2 above)
Comment (plaindocs) [inline user/for-beginners.md]: although it isn't just in the docs, it is also in the product. :-p

**Comment by plaindocs** [comment]
Hey @acnagy 

I might reword the error states so they are clearer and more parallel, 

```
+ * *errored* - a command returned a non-zero exit code in the `before_install`, `install`, or `before_script` section. The job stops immediately.
--
  |   | + * *failed* - a non-zero exit code is returned in the `script` section. The job continues to run until it completes.
  |   | + * *canceled* -  a user cancels the job before it completes.
```

Fix up the HVVM thing mentioned by @vagnerr in (https://github.com/travis-ci/docs-travis-ci-com/issues/1145#issuecomment-315043902), give this a quick once over, then merge it early next week, what do you think?

**Comment by plaindocs** [comment]
I'd also probably remove this section `Pre-getting started` and just make sure we're clear about the prerequisites in the Getting Started.

**Comment by plaindocs** [comment]
I'm not 100% happy with this getting started.

**Comment by acnagy** [comment]
@plaindocs I made a couple changes - look any better? 🤔

**Comment by plaindocs** [comment]
Hey Anna - just to be clear I was referring mostly to my work on the Getting Started guide up there, not to your work on the Beginners Guide.

**Comment by plaindocs** [comment]
Finally merging this, thanks for so many iterations!
